tests/test_metadata_handling.py

```diff
@@ -0,0 +1,80 @@
+import unittest
+import yaml
+from shaker import helpers
+
+class TestMetadataHandling(unittest.TestCase):
+    
+    # Sample metadata with duplicates
+    _sample_metadata_duplicates = \
+"""
+dependencies:
+    - git@github.com:test_organisation/test1-formula.git
+    - git@github.com:test_organisation/test1-formula.git
+    - git@github.com:test_organisation/test2-formula.git
+    - git@github.com:test_organisation/test3-formula.git
+    - git@github.com:test_organisation/test3-formula.git
+entry:
+    - dummy
+"""
+    _sample_metadata_no_duplicates = \
+"""
+dependencies:
+    - git@github.com:test_organisation/test1-formula.git
+    - git@github.com:test_organisation/test2-formula.git
+    - git@github.com:test_organisation/test3-formula.git
+entry:
+    - dummy
+"""
+    # Sample config that yaml load will fail to parse
+    _sample_metadata_bad_yaml = \
+    """
+    dependencies--
+        - git@github.com:test_organisation/test1-formula.git
+        - git@github.com:test_organisation/test1-formula.git
+        - git@github.com:test_organisation/test2-formula.git
+        - git@github.com:test_organisation/test3-formula.git
+        - git@github.com:test_organisation/test3-formula.git
+    entry--
+        - dummy
+    """
+    @classmethod
+    def setup_class(cls):
+        pass
+    
+    @classmethod
+    def teardown_class(cls):
+        pass
```

Comment:
Minor thing: these methods are optional and empty so could we remove them both please?